src/ucp/wireup/wireup.c

```diff
@@ -620,6 +623,73 @@ ucp_wireup_process_reply(ucp_worker_h worker, ucp_ep_h ep,
     }
 }
 
+static void ucp_wireup_close_flushed_callback(ucp_request_t *req)
+{
+    ucp_ep_h ep                = req->send.ep;
+
+    /* If a flush is completed from a pending/completion callback, we need to
+     * schedule slow-path callback to release the endpoint later, since a UCT
+     * endpoint cannot be released from pending/completion callback context.
+     */
+    ucs_trace("adding slow-path callback to destroy ep %p", ep);
+    req->send.disconnect.prog_id = UCS_CALLBACKQ_ID_NULL;
+    uct_worker_progress_register_safe(ep->worker->uct,
+                                      ucp_ep_local_disconnect_progress,
+                                      req, UCS_CALLBACKQ_FLAG_ONESHOT,
+                                      &req->send.disconnect.prog_id);
+}
+
+static UCS_F_NOINLINE void
+ucp_wireup_process_ep_check(ucp_worker_h worker, ucp_ep_h ep,
+                            const ucp_wireup_msg_t *msg,
+                            const ucp_unpacked_address_t *remote_address)
+{
+    ucs_status_t status;
+    ucp_ep_h reply_ep;
+    unsigned addr_indices[UCP_MAX_LANES];
+    ucs_status_ptr_t req;
+
+    if (ep != NULL) {
+        /* if EP exists then do not send reply */
+        return;
+    }
+
+    /* if endpoint does not exist - create a temporary endpoint to send a
+     * UCP_WIREUP_MSG_EP_REMOVED reply */
+    status = ucp_worker_create_ep(worker, 0, remote_address->name,
+                                  "wireup ep_check reply", &reply_ep);
+    if (status != UCS_OK) {
+        ucs_error("failed to create EP: %s", ucs_status_string(status));
+        goto out;
```

Comment:
since "out:" label does not do anything now, can replace by "return"

Reply:
ok, label is removed